### PR TITLE
Add kind picker to the CatalogPage

### PR DESCRIPTION
**Pull request by timbonicus**
Removes the hidden `EntityKindPicker` on the catalog index page and replaces it with a dropdown header to select entity kinds besides `Component`:

kind-picker

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by freben** [comment]
Cool stuff! I tried it out a bit.

- I see that the options matching in the picker is case sensitive. I got some warnings when on `?filters%5Bkind%5D=template` for example; the page just breaks and there are console warnings. It should be case insensitive.
- Likewise, if you give it a kind that isn't used by any entity, then it also ends up rendering absolutely nothing (you just see the little down chevron) and get console warnings. You should probably add the fallback behavior of always adding the filter-input as a candidate in the dropdown if it doesn't already exist. That way, if you go to `?filters%5Bkind%5D=Foo` then at least the dropdown renders with "Foo" in it and there are no matches as expected.
- This also makes me wonder how it behaves with a completely empty catalog (haven't tried).
- The `Name` column has a fixed default kind, so if I select to list users, they are all shown as `user:foo`. Could you make that dynamic as well?

File: plugins/catalog/src/components/CatalogKindHeader/CatalogKindHeader.test.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  {
+    apiVersion: '1',
+    kind: 'System',
+    metadata: {
+      name: 'system',
+    },
+  },
+];
+
+const apis = ApiRegistry.from([
```
Comment (freben) [inline plugins/catalog/src/components/CatalogKindHeader/CatalogKindHeader.test.tsx:61]: hint: the `ApiRegistry.with(a, b).with(c, d)` syntax is often more convenient (and prettifies nicer) than `.from([[a, b], [c, d]])`

File: plugins/catalog/src/components/CatalogKindHeader/CatalogKindHeader.test.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  [
+    catalogApiRef,
+    {
+      getEntities: jest
+        .fn()
+        .mockImplementation(() => Promise.resolve({ items: entities })),
```
Comment (freben) [inline plugins/catalog/src/components/CatalogKindHeader/CatalogKindHeader.test.tsx:67]: hint: it's a bit terser with

```ts
jest.fn(async () => x);
```

or

```ts
jest.fn().mockResolvedValue(x);
```

**Review by freben — APPROVED** [review]
NICE! 🥳

**Comment by timbonicus** [comment]
Well those suggestions were just lovely, much obliged @freben ❤️‍🔥

File: .changeset/nasty-ads-nail.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+'@backstage/core-components': minor
+---
+
+Changed the `titleComponent` prop on `ContentHeader` to accept `JSX.Element` instead of a React `ComponentType`. Usages of this prop should be converted from passing a component to passing in the rendered element:
```
[the remark this answers is not in the document]
Comment (timbonicus) [reply .changeset/nasty-ads-nail.md:5]: This does look slightly more broad, updated! 🙇

**Pull request merged**